forge-gui/res/cardsfolder/upcoming/sarinth_steelseeker.txt

```diff
@@ -0,0 +1,12 @@
+Name:Sarinth Steelseeker
+ManaCost:1 G
+Types:Creature Human Artificer Scout
+PT:1/2
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Artifact.YouCtrl | TriggerZones$ Battlefield | Execute$ TrigDig | TriggerDescription$ Whenever an artifact enters the battlefield under your control, look at the top card of your library. If it's a land card, you may reveal it and put it into your hand. If you don't put the card into your hand, you may put it into your graveyard.
+SVar:TrigDig:DB$ PeekAndReveal | PeekAmount$ 1 | RevealValid$ Land | RevealOptional$ True | RememberRevealed$ True | SubAbility$ DBChangeZone
+SVar:DBChangeZone:DB$ ChangeZone | Defined$ TopOfLibrary | Origin$ Library | Optional$ True | Destination$ Hand | ConditionDefined$ Remembered | ConditionPresent$ Land | ConditionCompare$ EQ1 | SubAbility$ DBGrave
```

Comment:
This seems wrong... it you were able to reveal/imprint the card in line 6, we don't need to check that it's a Land still here... probably `Defined$ Imprinted | ForgetChanged$ True` and remove Optional and all the Condition stuff

The ForgetChanged would set up the next line

Reply:
Why remove optional ?